packages/core/echo/indexing/src/types.ts

```diff
@@ -17,10 +17,13 @@ export type IndexQuery = {
    * null means all Expando objects.
    * undefined means all objects (no filter).
    */
-  typename?: string | null;
+  typenames: (string | null | undefined)[];
```

Comment:
does it make sense to include `undefiend` as one possible array value

Reply:
To support `db.query()` (query all objects)

Reply:
what would a filter of `["example/Type", undefined]` do? query everying?